src/utils/calendarEvents.ts

```diff
@@ -85,26 +86,74 @@ export const useColors = (fieldConfig?: FieldConfigSource) => {
   }, [fieldConfig?.defaults.color, theme]);
 };
 
+/**
+ * Get Minutes Offset From Time Zone
+ * @param timeZone
+ */
+export const getMinutesOffsetFromTimeZone = (timeZone: TimeZone) => {
+  if (timeZone === InternalTimeZones.localBrowserTime) {
+    /**
+     * Offset is not needed, dates are in browser time zone
+     */
+    return 0;
+  }
+
+  /**
+   * Calculate offset to show date in dashboard time zone for user
+   */
+  if (timeZone === InternalTimeZones.utc) {
+    /**
+     * UTC offset from browser date
+     */
+    return new Date().getTimezoneOffset();
+  }
+
+  const date = new Date();
+
+  /**
+   * Browser Date
+   * Reset milliseconds to prevent losing 1 minute in difference
+   */
+  const browserDate = dayjs(date).set('milliseconds', 0);
+
+  /**
+   * Time Zone Date
+   */
+  const timeZoneDate = dayjs(date.toLocaleString('en-US', { timeZone }));
```

Comment:
It should be set to a selected locale.

Reply:
the date is being used only for timezone offset calculation so we can use any locale here